### ISSUE TITLE
May be some errors! 

### ISSUE THREAD

**Issue by qqsuhao** [eriklindernoren/PyTorch-YOLOv3#652]
Thanks for the author's code.
I find some error about the code, but I am not sure.
-
- On the line 199 of file "yolov3-tiny.cfg", it should be "mask=0,1,2".
- On the line 329 of file "utils/utils.py", I find that if use ".float()" to convert the BoolTensor to the FloatTensor, the bool values false and true can not be converted as 0 and 1. It should be revised as follows 
 `tconf = FloatTensor(nB, nA, nG, nG).fill_(0)`
 `tconf[obj_mask] = 1`

**Comment by Flova** [comment]
Thanks for your issue.

- Regarding your first point. This is known and discussed in other issues. The problem is that the original darknet implementation is wrong at this point and we want to use the pretrained weights.

- Could you further explain what's wrong with the `.float()` cast? Afaik it works and a quick search showed multiple pytorch examples showing that cast.

**Comment by qqsuhao** [comment]
Thanks for your reply.
For the sencond issue, I find that using `.float()` to convert BoolTensor may get wrong results, such as follows:
![image](https://user-images.githubusercontent.com/31772922/112329133-e44a1780-8cf1-11eb-9813-41e05d6b60d9.png)

**Comment by Flova** [comment]
Oh, this seems indeed like uninitialized memory or something similar.
Did you mean the one in line 290?

**Comment by qqsuhao** [comment]
Oh, sorry, I make a mistake in the first comment. 
It is the one in line 290 but not 329.

**Comment by Flova** [comment]
Could you provide a pr for this? :)

**Comment by Flova** [comment]
I think the current version only casts ints to float. I therefore would close this issue.
